geo-traits/src/dimension.rs

```diff
@@ -0,0 +1,34 @@
+/// The logical dimension of the geometry.
+///
+///
+#[allow(clippy::upper_case_acronyms)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum Dimensions {
+    /// A two-dimensional geometry with X and Y values
+    XY,
```

Comment:
Non-blocking: What do you think about using the `Xyz` capitalization convention? It's acronym-ish, which means [this suggestion](https://github.com/rust-lang/rfcs/blob/master/text/0430-finalizing-naming-conventions.md#fine-points) would be relevant:

> In `UpperCamelCase`, acronyms count as one word: use `Uuid` rather than
`UUID`.  In `snake_case`, acronyms are lower-cased: `is_xid_start`.

Reply:
And if so we could remove the `#[allow(clippy::upper_case_acronyms)]